cms/static/js/factories/login.js

```diff
@@ -22,15 +22,19 @@ define(['jquery.cookie', 'utility'], function() {
 
         $('form#login_form').submit(function(event) {
             event.preventDefault();
+            var submit_button = $('#submit'),
+                deferred = new $.Deferred(),
+                promise = deferred.promise();
+            ViewUtils.disableElementWhileRunning(submit_button, function() { return promise; });
```

Comment:
Nit: this is just a stylistic thing, but this function was intended to wrap around the call in question. I think it is easier to read in that case, rather than having a separate function just to return the promise. i.e. something like:

    ViewUtils.disableElementWhileRunning(submit_button, function() {
        var submit_data = $('#login_form').serialize();
        ....
    });